adafruit_bluefruit_connect/button_packet.py

```diff
@@ -0,0 +1,89 @@
+# The MIT License (MIT)
+#
+# Copyright (c) 2019 Dan Halbert for Adafruit Industries
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+"""
+`adafruit_bluefruit_connect.controller_packet`
+====================================================
+
+Bluefruit Connect App Button data packet (button_name, pressed/released)
+
+
+* Author(s): Dan Halbert for Adafruit Industries
+
+"""
+
+import struct
+
+from .packet import Packet
+
+class ButtonPacket(Packet):
+    """A packet containing a button name and its state"""
+
+    _FMT_PARSE = '<xxssx'
+    PACKET_LENGTH = struct.calcsize(_FMT_PARSE)
+    # _FMT_CONSTRUCT doesn't include the trailing checksum byte.
+    _FMT_CONSTRUCT = '<2sss'
+    _TYPE_HEADER = b'!B'
+
+    def __init__(self, button, pressed):
+        """Construct a ButtonPacket from a button name and the button's state.
+
+        :param str button: a single character denoting the button
+        :param bool pressed: ``True`` if button is pressed; ``False`` if it is release.
+        """
+        # This check will catch wrong length and also non-sequence args (like an int).
+        try:
+            assert len(button) == 1
+        except:
+            raise ValueError("Button must be a single char or byte.")
+
+        self._button = button
+        self._pressed = pressed
+
+    @classmethod
+    def parse_private(cls, packet):
```

Comment:
Did you try the name `_parse`? _ is usually used to designate private but I have a feeling lint may have complained about it.

Reply:
Yes, it was a pylint problem, because I was not calling it as `cls._parse()`. I could use the `generated-name` pylint exclusion, but that's kind of misleading. I saw other people complain about this on stackoverflow as well.